_studio/mfx_lib/encode_hw/vp9/include/mfx_vp9_encode_hw.h

```diff
@@ -58,6 +58,8 @@ class MFXVideoENCODEVP9_HW : public VideoENCODE
 
         if (core)
             m_pCore = core;
+        else
+            return MFX_ERR_NULL_PTR;
```

Comment:
no. It is constructor, it can't return status. Fix it in the way you did in previous commit - place default initialization to the list:
```
        : m_bStartIVFSequence(false)
        , m_maxBsSize(0)
        , m_pCore(nullptr) // <-- !!!
        , m_initialized(false)
        , m_frameArrivalOrder(0)
        , m_drainState(false)
        , m_resetBrc(false)
        , m_taskIdForDriver(0)
        , m_numBufferedFrames(0)
        , m_initWidth(0)
        , m_initHeight(0)
        , m_frameOrderInGop(0)
        , m_frameOrderInRefStructure(0)
```

Reply:
even more, you can put initialization in the list explicitly. This check of pointer in constructor body makes no sense for me:
```
, m_pCore(core)
```

Reply:
@mgonchar 
Hi, would you mind give a review to this pr? Thank you so much. Really appreciate.